src/BladeSDK.ts

```diff
@@ -141,27 +152,30 @@ export class BladeSDK {
         this.apiKey = apiKey;
         this.network = StringHelpers.stringToNetwork(network);
         this.dAppCode = dAppCode;
-        this.visitorId = visitorId;
         this.sdkEnvironment = sdkEnvironment;
         this.sdkVersion = sdkVersion;
 
         initApiService(apiKey, dAppCode, sdkEnvironment, sdkVersion, this.network, visitorId);
-        try {
-            visitorId = await decrypt(localStorage.getItem("BladeSDK.visitorId") || "", this.apiKey);
-        } catch (e) {
-            // console.log("failed to decrypt visitor id", e);
+        if (!visitorId) {
+            try {
+                visitorId = await decrypt(localStorage.getItem("BladeSDK.visitorId") || "", this.apiKey);
```

Comment:
Are you reassigning argument here?

Reply:
It's good that you are not changing properties of an object argument, but as a good practice you should still avoid of doing so.

Reply:
fixe